### PR TITLE
vault module

### PR DESCRIPTION
**Pull request by Polina-Gubina**

### CODE REVIEW

File: plugins/modules/cbr_vault.py [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+            return True
+        if state == 'absent' and vault:
+            return True
+        return False
+
+    def _require_update(self, vault):
+        require_update = False
+        if vault:
+            if self.params['billing']:
+                if self.params['billing']['size'] != vault['billing']['size']:
```
Comment (vladimirvshivkov) [inline plugins/modules/cbr_vault.py:450]: no need for double ifs, leave only  `if self.params['billing']['size'] != vault['billing']['size']:`

**Review by vladimirvshivkov — CHANGES_REQUESTED** [review]

File: plugins/modules/cbr_vault.py [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+      - What needs to be done.
+    choices: ['associate_resources', 'dissociate_resources', 'bind_policy',\
+                                                                'unbind_policy']
+    type: str
+requirements: ["openstacksdk", "otcextensions"]
+'''
+
+RETURN = '''
+cbr_vault:
+    description: AS groups object.
```
Comment (vladimirvshivkov) [inline plugins/modules/cbr_vault.py:197]: maybe vault object?

File: plugins/modules/cbr_vault.py [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+          - Value. It is mandatory when a tag is added and optional when\
+            a tag is deleted.
+        type: str
+  auto_bind:
+    description:
+      - Whether automatic association is supported.
+    type: bool
+  name:
+    description:
+      - Used for modifying.
```
[the remark this answers is not in the document]
Comment (anton-sidelnikov) [reply plugins/modules/cbr_vault.py:139]: we do not modifying names in modules

File: plugins/modules/cbr_vault.py [hunk 4]
```diff
@@ -0,0 +1,554 @@
...
+
+        if state == 'absent':
+            if self.ansible.check_mode:
+                self.exit_json(changed=False)
+
+        if action in ('associate_resources', 'dissociate_resources',
+                      'bind_policy', 'unbind_policy') or state == 'absent':
+            if self.ansible.check_mode:
+                self.exit_json(changed=False)
+            self.fail_json('vault not found')
```
Comment (anton-sidelnikov) [inline plugins/modules/cbr_vault.py:535]: add vault id or name in message

File: plugins/modules/cbr_vault.py [hunk 5]
```diff
@@ -0,0 +1,554 @@
...
+                if self.params['name']:
+                    attrs['name'] = self.params['name']
+                self.conn.cbr.update_vault(vault=vault, **attrs)
+                self.exit(changed=True)
+
+            if state == 'absent':
+                self.conn.cbr.delete_vault(vault=vault)
+                self.exit(changed=True)
+
+        if state == 'absent':
```
Comment (anton-sidelnikov) [inline plugins/modules/cbr_vault.py:527]: add some message or just remove this

File: tests/integration/targets/cbr/tasks/main.yaml [hunk 6]
```diff
@@ -0,0 +1,102 @@
+---
```
Comment (anton-sidelnikov) [inline tests/integration/targets/cbr/tasks/main.yaml:1]: do not run this test in suite, add alias with disabled state

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: tests/integration/targets/cbr/aliases [hunk 7]
```diff
@@ -0,0 +1 @@
+disabled
```
Comment (anton-sidelnikov) [inline tests/integration/targets/cbr/aliases:1]: add empty line

**Review by anton-sidelnikov — APPROVED** [review]

File: plugins/modules/cbr_vault.py [hunk 8]
```diff
@@ -0,0 +1,556 @@
...
+      - What needs to be done.
+    choices: ['associate_resources', 'dissociate_resources', 'bind_policy',\
+                                                                'unbind_policy']
+    type: str
+requirements: ["openstacksdk", "otcextensions"]
+'''
+
+RETURN = '''
+vault:
+    description: AS groups object.
```
Comment (vladimirvshivkov) [inline plugins/modules/cbr_vault.py:193]: unchanged

**Review by vladimirvshivkov — CHANGES_REQUESTED** [review]

**Review by vladimirvshivkov — APPROVED** [review]

**Pull request merged**